File: backend/src/main/java/com/musclemap/workout/SplitType.java
package com.musclemap.workout;

/**
 * Training split for a generated program. Persisted as name()
 * (generated_programs.split_type CHECK constraint). Mirrors the frontend split presets.
 */
public enum SplitType {
    FULL_BODY,
    UPPER_LOWER,
    PUSH_PULL_LEGS,
    BRO_SPLIT,
    BODY_PART
}

File: frontend/src/domain/enums/SplitType.ts
/** The training split a generated program is organised around. */
export enum SplitType {
  FullBody = 'FULL_BODY',
  UpperLower = 'UPPER_LOWER',
  PushPullLegs = 'PUSH_PULL_LEGS',
  /** Classic body-part / "bro" split: Chest+Triceps, Back+Biceps, Legs, Shoulders+Core. */
  BodyPart = 'BODY_PART',
}
